example/test.py

```diff
@@ -20,14 +22,14 @@ def wait_for_tx(cl: Client, hash: str):
     listener = cl.listen(filter)
 
     for event in listener:
-        if event["Pipeline"]["hash"] == hash:
-            if event["Pipeline"]["status"] == "Committed":
-                return
-            elif event["Pipeline"]["status"] == "Validating":
-                pass
-            else:
-                raise RuntimeError(
-                    f"Tx rejected: {event['Pipeline']['status']}")
+        if event.variant is event.Type.Pipeline:
+            if event.value.hash == hash:
+                if event.value.status.variant is pipeline.Status.Type.Committed:
+                    return
+                elif event.value.status.variant is pipeline.Status.Type.Validating:
+                    pass
+                else:
+                    raise RuntimeError(f"Tx rejected: {event.value.status}")
```

Comment:
This change isn't particularly pythonic. But it's a step in the right direction.

Reply:
Yeah, here I'm mostly focusing on at least having actual types instead of jsonly-typed blobs of something.
Making it look like Python will require either way more future work or some approach to codegen I didn't think of.

Reply:
We should discuss codegen.